pyopenms_viz/testing/BokehSnapshotExtension.py

```diff
@@ -38,136 +40,149 @@ def handle_data(self, data):
-                    print(f"Key {key} not in second json")
+                    logger.warning(f"Key {key} not found in the second JSON.")
                     return False
-                elif key in ["id", "root_ids"]:  # add keys to ignore here
-                    pass
+                elif key in ["id", "root_ids"]:  # Keys to ignore
+                    continue
                 elif not BokehSnapshotExtension.compare_json(json1[key], json2[key]):
-                    print(f"Values for key {key} not equal")
+                    logger.warning(f"Values for key {key} are not equal.")
                     return False
             return True
+
         elif isinstance(json1, list) and isinstance(json2, list):
             if len(json1) != len(json2):
-                print("Lists have different lengths")
+                logger.warning("Lists have different lengths.")
                 return False
-            # lists are unordered so we need to compare every element one by one
-            for idx, i in enumerate(json1):
-                check = True
-                if isinstance(i, dict):
-                    if (
-                        "type" not in i.keys()
-                    ):  # if "type" not present than dictionary with only id, do not need to compare, will get key error if check
-                        check = False
-                        pass
-                    if check:  # find corresponding entry in json2 only if check is true
-                        for j in json2:
-                            if (
-                                "type" not in j.keys()
-                            ):  # if "type" not present than dictionary only has id, do not need to compare, will get key error if check
-                                check = False
-                            if check and (j["type"] == i["type"]):
-                                if not BokehSnapshotExtension.compare_json(i, j):
-                                    print(f"Element {i} not equal to {j}")
-                                    return False
-                                return True
-                        print(f"Element {i} not in second list")
-                        return False
+
+            matched_elements = []
+            for i in json1:
+                matched = any(
+                    BokehSnapshotExtension.compare_json(i, j)
+                    for j in json2 if j not in matched_elements
+                )
+                if matched:
+                    matched_elements.append(i)
                 else:
-                    return json1[idx] == json2[idx]
+                    logger.warning(f"Element {i} not found in the second list.")
+                    return False
             return True
+
         else:
             if json1 != json2:
-                print(f"Values not equal: {json1} != {json2}")
+                logger.warning(f"Values not equal: {json1} != {json2}")
             return json1 == json2
```

Comment:
The logic of this function might be changing. Seems that tests for this now fail